x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityCcsWithMultipleRemotesRestIT.java

```diff
@@ -0,0 +1,196 @@
+                    randomFrom("cluster1_index1", "*_index1", "*"),
+                    randomFrom("cluster2_index1", "*_index1", "*"),
+                    randomBoolean()
+                ),
+                "cluster1_index1",
+                "cluster2_index1"
+            );
+
+            // Search single remote
+            final boolean searchFirstCluster = randomBoolean();
+            final String expectedIndex = searchFirstCluster ? "cluster1_index1" : "cluster2_index1";
+            searchAndAssertIndicesFound(
+                String.format(
+                    Locale.ROOT,
+                    "/%s:%s/_search?ccs_minimize_roundtrips=%s",
+                    searchFirstCluster ? "my_remote_cluster" : "my_remote_cluster_2",
+                    randomFrom(expectedIndex, "*_index1", "*"),
+                    randomBoolean()
+                ),
+                expectedIndex
+            );
+
+            // Unauthorized access on either of the remotes fails the whole request
+            // No permissions for this index name, so searching for it on either remote will result in 403
+            final String missingIndex = "index1";
+            final boolean missingIndexOnFirstCluster = randomBoolean();
+            // Make sure we search for missing index on at least one remote, possibly both
+            final boolean missingIndexOnSecondCluster = false == missingIndexOnFirstCluster || randomBoolean();
+            final ResponseException exception1 = expectThrows(
+                ResponseException.class,
+                () -> performRequestWithRemoteSearchUser(
+                    new Request(
+                        "GET",
+                        String.format(
+                            Locale.ROOT,
+                            "/my_remote_cluster:%s,my_remote_cluster_2:%s/_search?ccs_minimize_roundtrips=%s",
+                            missingIndexOnFirstCluster ? missingIndex : randomFrom("cluster1_index1", "*_index1", "*"),
+                            missingIndexOnSecondCluster ? missingIndex : randomFrom("cluster2_index1", "*_index1", "*"),
+                            randomBoolean()
+                        )
+                    )
+                )
+            );
+            assertThat(exception1.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+
+            // Same as above but with wildcard cluster alias
+            final ResponseException exception2 = expectThrows(
+                ResponseException.class,
+                () -> performRequestWithRemoteSearchUser(
+                    new Request(
+                        "GET",
+                        String.format(Locale.ROOT, "/*:%s/_search?ccs_minimize_roundtrips=%s", missingIndex, randomBoolean())
+                    )
+                )
+            );
+            assertThat(exception2.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+        }
```

Comment:
There are likely more scenarios here. Let me know if there's ones you think are a must-cover for this PR. Otherwise, I would prefer to add these iteratively, in follow ups since this PR is already quite big.